src/widget_type.h

```diff
@@ -665,14 +665,14 @@ class Scrollbar {
 	 * @param num the number of elements in the list
 	 * @note updates the position if needed
 	 */
-	void SetCount(int num)
+	void SetCount(uint num)
 	{
-		assert(num >= 0);
-		assert(num <= MAX_UVALUE(uint16));
-
 		this->count = num;
-		num -= this->cap;
-		if (num < 0) num = 0;
+		if (this->cap > num) {
```

Comment:
To come back to `>` vs `<`, this was the one that triggered me most. Here:

```
if (num < this->cap) {
  num = 0;
} else {
  num -= this->cap;
}
```

Reads much more natural to me. But again, this can completely be because my brain is primed like that, I am just mentioning it so you can consider it :)